apps/extension/src/ui/apps/dashboard/routes/Settings/Mnemonics/MnemonicBackupModal/ModalRouter.tsx

```diff
@@ -0,0 +1,31 @@
+/* eslint-disable react/display-name */
+import { useMemo } from "react"
+import { Modal } from "talisman-ui"
+
+import { Acknowledgement } from "./Acknowledgement"
+import { Complete } from "./Complete"
+import { Stages, useMnemonicBackupModal } from "./context"
+import { ShowMnemonic } from "./Show"
+
+export const MnemonicBackupModalRouter = () => {
+  const { close, isOpen, stage } = useMnemonicBackupModal()
+
+  const Component = useMemo(() => {
+    switch (stage) {
+      case Stages.Acknowledgement:
+        return Acknowledgement
+      case Stages.Show:
+        return ShowMnemonic
+      case Stages.Complete:
+        return Complete
+      default:
+        return Acknowledgement
+    }
+  }, [stage])
+
+  return (
+    <Modal containerId="main" isOpen={isOpen} onDismiss={close}>
+      {Component && <Component />}
+    </Modal>
+  )
+}
```

Comment:
Trying out the code suggestions thingy, but I think it messed up the diff:

```suggestion
import { useMemo } from "react"
import { Modal } from "talisman-ui"

import { Acknowledgement } from "./Acknowledgement"
import { Complete } from "./Complete"
import { Stages, useMnemonicBackupModal } from "./context"
import { ShowMnemonic } from "./Show"

export const MnemonicBackupModalRouter = () => {
  const { close, isOpen, stage } = useMnemonicBackupModal()

  const Component = useMemo(() => {
    switch (stage) {
      case Stages.Acknowledgement:
        return Acknowledgement
      case Stages.Show:
        return ShowMnemonic
      case Stages.Complete:
        return Complete
      default:
        return Acknowledgement
    }
  }, [stage])

  return (
    <Modal containerId="main" isOpen={isOpen} onDismiss={close}>
      {Component && <Component />}
    </Modal>
  )
}
MnemonicBackupModalRouter.displayName = "MnemonicBackupModalRouter"
```